### PR TITLE
Prettify chapter2

### PR DESCRIPTION
**Pull request by jitka1997**
Mostly modified and hopefully prettified sections:
- Hadamard matrix
- symmetric block design

Also partially resolved TODOS at the end of difference sets section

### CODE REVIEW

File: block_designs.tex [hunk 1]
```diff
@@ -1,6 +1,144 @@
 \chapter{Blokové plány}
 
+Blokové plány sú kombinatorické štruktúry, pre ktoré existuje veľa využití v štatistike, dizajne experimentov, teórii kódovania, teórii grafov a ďalších oblastiach. Napríklad aj pri dizajne spoločenských hier :) Populárna hra Dobble je vlastne blokový plán, kde jednotlivé karty sú bloky a symboly na kartách sú objekty.
+
+\section{Symetrické blokové plány}
+
+\begin{definition}
+
+Nech $X = \set{x_1, \ldots, x_v}$ je množina objektov a $\mathcal{B} = \set{X_1, \ldots, X_v} \subset \powerset(X)$ je množina podmnožín objektov (tieto podmnožiny voláme \emph{bloky}), pričom sú splnené nasledujúce podmienky:
```
Comment (dodo42) [inline block_designs.tex:9]: If I scanned through the files correctly, `\subseteq` is used for subsets

File: block_designs.tex [hunk 2]
```diff
@@ -1,6 +1,144 @@
...
+\begin{enumerate}
+	\item $|X_i| = k$ pre $i = 1, 2, \ldots, v$
```
Comment (dodo42) [inline block_designs.tex:11]: There should be semicolons (;) at the end of the items and dot at the end of the last item.

File: block_designs.tex [hunk 3]
```diff
@@ -1,6 +1,144 @@
...
+	\item $|X_i \cap X_j| = \lambda$ pre $i \neq j$
+	\item $0 < \lambda < k < v - 1$
+\end{enumerate}
+Potom systém blokov $\mathcal{B}$ voláme $(v, k, \lambda) - konfigurácia$ (alebo symetrický blokový plán).
```
Comment (dodo42) [inline block_designs.tex:15]: Dolars should be used for math, not for italits. For this purpose, use `\emph{$(v, k, \lambda)$-konfigurácie}`

File: block_designs.tex [hunk 4]
```diff
@@ -1,6 +1,144 @@
...
+
+\begin{proof}
+Dokážeme postupne jednotlivé body.
+\begin{enumerate}
+	\item $A J = k J$. \label{itemA} \\
+	Riadok $i$ matice $A$ popisuje, ktoré objekty tvoriace $i$-ty blok. Všetky objekty v $i$-tom bloku sú zastúpené v $i$-tom riadku matice $A$ jednotkami, ostatné objekty sú zastúpené nulami. Keďže každý blok má $k$ prvkov a maticu $J$ tvoria samé jednotky. Tak prvok matice $A J$ vznikne ako súčet $k$ jednotiek a $v - k$ núl, teda každý prvok výslednej matice má hodnotu $k$ a $A J = k J$.
+
+	\item $A A^T = \lambda J + (k - \lambda) I$. \label{itemB} \\
+	Prvok $A A^T$ na pozícii $i, j$ označme $c_{i,j}$ potom $c_{i,j} = \sum\limits_{l=1}^{v} a_{il} a_{jl}$. Keďže $a_{pq}$ je jedna v prípade, že $x_q \in X_p$, inak $a_{pq} = 0$, tak súčin $a_{il} a_{jl} = 1$ ak $x_l \in X_i \cap X_j$, inak $a_{il} a_{jl} = 0$. Z toho vyplýva, že:
+	\begin{minipage}{.5\linewidth}
```
Comment (dodo42) [inline block_designs.tex:47]: I'd personally remove the minipage and leave the equation on a single line.

File: block_designs.tex [hunk 5]
```diff
@@ -1,6 +1,144 @@
...
+	Teda $A A^T = \begin{pmatrix}
+		k & \lambda & \dots & \lambda \\
+		\lambda & \ddots & & \vdots \\
+		\vdots &  & \ddots & \lambda \\
+		\lambda & \dots & \lambda & k \\
+	\end{pmatrix} =  \lambda J + (k - \lambda) I$.
+
+	\item $det(A A^T) = (det(A))^2 = (k + \lambda (v - 1)) (k - \lambda)^{v-1}$. \label{itemC} \\
+
+	$ det(A A^T) = \begin{vmatrix}
```
Comment (dodo42) [inline block_designs.tex:64]: `\det`

File: block_designs.tex [hunk 6]
```diff
@@ -1,6 +1,144 @@
...
+		0 & 0 & \dots & 0 & k - \lambda \\
+	\end{vmatrix} = (\lambda (v - 1) + k) (k - \lambda)^{v-1}
+	$
+
+	\item $k (k-1) = \lambda (v-1)$. \label{itemD} \\
+	Vychádzajme z rovnosti $A A^T = \lambda J + (k - \lambda) I$. Ďalej počítajme:
+	$$ A A^T = \lambda J + (k - \lambda) I \quad / \dot J$$
+	$$ A A^T J = \lambda J^2 + (k - \lambda) J = \lambda v J + (k - \lambda) J = (k + \lambda(v - 1)) J $$
+	Pretože $0 < \lambda < k$ je podľa \ref{itemC} matica $A$ regulárna a podľa \ref{itemA} platí $A^{-1} J = \frac{1}{k} J$. Z toho a rovnosti $A A^T J = (k + \lambda(v - 1)) J$ vyplýva $A^T J = (k + \lambda (v - 1)) A^{-1} J$
+	$$ A^T J = \frac{k + \lambda (v - 1)}{k} J \quad / transponujeme$$
```
Comment (dodo42) [inline block_designs.tex:97]: You may use `\text{transponujeme}`, so it is not italics.

File: block_designs.tex [hunk 7]
```diff
@@ -1,6 +1,144 @@
...
+	\end{equation*}
+	\begin{equation*}
+		\lambda A^{-1} J A + (k - \lambda) I \overset{\ref{itemE}}{=} \lambda A^{-1} k J + (k - \lambda) I \overset{\ref{itemA}}{=} \lambda A^{-1} A J + (k - \lambda) I = \lambda J + (k - \lambda) I = A A^T
+	\end{equation*}
+\end{enumerate}
+\end{proof}
+
+\begin{remark}
+	Tvrdenie z vlastnosti \ref{itemE} sa dá ekvivalentne zapísať ako $\sum\limits_{i=1}^{v} a_{ij} = k$, inými slovami súčet prvkov v stĺpci matice $A$ je $k$, teda každý objekt je obsiahnutý v k blokoch. Podobne z vlastnosti \ref{itemF} vyplýva, že: 
+	\begin{minipage}{.5\linewidth}
```
Comment (dodo42) [inline block_designs.tex:124]: Also this can be on its own line.

File: block_designs.tex [hunk 8]
```diff
@@ -263,20 +401,26 @@ \section{Cyklické blokové plány a diferenčné množiny}
...
+	
+	$$v = p = 4x^2+1, k = x^2, \lambda = \frac{x^2 - 1}{4}$$
+	
+	\end{theorem_hard}
 
-\todo[inline]{rozpisat bikvadraticke rezidua (neviem to najst v literature)}
-\todo[inline]{rozpisat twin prime power (neviem to najst v literature)}
-\end{toreview}
+\begin{theorem_hard}{(Typ T dif. množín --- Twin prime power dif. množiny)}\\ 
+	Nech je množina $D$ v grupe $((GF(q), +) \times (GF(q + 2), +), +)$, kde $q$ aj $q + 2$ sú mocniny prvočísel, definovaná nasledovne $D = \{(x, y) ~|~ y = 0$ alebo x aj y sú nunulové a oba sú súčasne štvorce alebo súčasne neštvorce$\}$. Potom $D$ je $(v, k, \lambda)$-diferenčná množina s parametrami:
```
Comment (dodo42) [inline block_designs.tex:415]: nunulové  -> nenulové

File: block_designs.tex [hunk 9]
```diff
@@ -263,20 +401,26 @@ \section{Cyklické blokové plány a diferenčné množiny}
...
+	$$v = q^2 + 2q, k = \frac{q^2 + 2q - 1}{2}, \lambda = \frac{q^2 + 2q - 3}{4}$$
```
Comment (dodo42) [inline block_designs.tex:416]: It'd be nice to add spaces or maybe even `\quad` after commas.

File: block_designs.tex [hunk 10]
```diff
@@ -263,20 +401,26 @@ \section{Cyklické blokové plány a diferenčné množiny}
...
-
-\end{remark}
-\begin{toreview}
+\begin{theorem_hard}{(Typ B dif. množín --- bikvadratické rezíduá)}\\
+	Nech $F := GF(q)$ je konečné pole, kde $q = p^l$ a $p = 4x^2 + 1$ pre nepárne $x$. 
+	Nech $r$ je generátor grupy $F = (GF(q), +)$. 
+	Potom množina bikvadratických rezíduí grupy $F$ $BQR(F) := \set{TODO}$
+	je $(v, k, \lambda)$-diferenčnou množinou s parametrami:
+	
+	$$v = p = 4x^2+1, k = x^2, \lambda = \frac{x^2 - 1}{4}$$
```
Comment (dodo42) [inline block_designs.tex:410]: It'd be nice to add spaces or maybe even `\quad` after commas.

File: block_designs.tex [hunk 11]
```diff
@@ -263,20 +401,26 @@ \section{Cyklické blokové plány a diferenčné množiny}
...
-\begin{remark}
-
-Existujú aj ďalšie triedy diferenčných množín, napríklad bikvadratické rezíduá alebo tzv. \emph{twin prime power difference set}.
-
-\end{remark}
-\begin{toreview}
+\begin{theorem_hard}{(Typ B dif. množín --- bikvadratické rezíduá)}\\
+	Nech $F := GF(q)$ je konečné pole, kde $q = p^l$ a $p = 4x^2 + 1$ pre nepárne $x$. 
+	Nech $r$ je generátor grupy $F = (GF(q), +)$. 
+	Potom množina bikvadratických rezíduí grupy $F$ $BQR(F) := \set{TODO}$
```
Comment (dodo42) [inline block_designs.tex:407]: This TODO is intended?

**Review by dodo42 — CHANGES_REQUESTED** [review]
Nice work. I haven't study proofs much, only some of them briefly. There are some minor typos and I sugeest some TeX- or formatting enhancement. You don't need to correct all of them, since these are only notes for students, where formatting does not matter so much.

File: block_designs.tex (hunk 3 above)
Comment (jitka1997) [reply block_designs.tex:15]: Removed italics from "(v, k, \lambda) - konfigurácia" notation everywhere apart definition to be in accordance with "(v, k, \lambda)-diferenčnou množinou" notation.

File: block_designs.tex (hunk 5 above)
Comment (jitka1997) [reply block_designs.tex:64]: fixed every occurrence

File: block_designs.tex (hunk 9 above)
Comment (jitka1997) [reply block_designs.tex:416]: fixed in every diff. set type

File: block_designs.tex (hunk 11 above)
Comment (jitka1997) [reply block_designs.tex:407]: Yes, I didn't manage to write a correct and pretty definition of this set.

**Review by dodo42 — APPROVED** [review]

**Pull request merged**